### ISSUE TITLE
Error when invoking doesFeatureMatch in the support file after v15 release

### ISSUE THREAD

**Issue by apehead** [badeball/cypress-cucumber-preprocessor#1025]
### Current behavior

When using the `isFeature` and `doesFeatureMatch` functions, as explained in the documentation found at [this link](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/mixing-types.md), within the support file (e.g., `cypress/support/e2e.js`) like this:

```js
import {
  isFeature,
  doesFeatureMatch
} from '@badeball/cypress-cucumber-preprocessor';

beforeEach(() => {
  // This is only run for Cucumber-type specs
  if (isFeature() && doesFeatureMatch('@log')) {
    console.log('*** LOG ***');
  }
});
```

The following error is observed:

```
Expected to find internal properties, but didn't. This is likely because you're calling doesFeatureMatch() in a non-feature spec. Use doesFeatureMatch() in combination with isFeature() if you have both feature and non-feature specs (this might be a bug, please report at https://github.com/badeball/cypress-cucumber-preprocessor)

Because this error occurred during a before each hook we are skipping all of the remaining tests.
``` 

![cypress-cucumber-issue](https://github.com/badeball/cypress-cucumber-preprocessor/assets/6587503/3d4cc9e4-313c-447d-8713-d99b7fcb8ecf)

### Desired behavior

Using the `isFeature` and `doesFeatureMatch` functions in the support file (e.g., `cypress/support/e2e.js`), as in the following example, should not result in an error:

```js
import {
  isFeature,
  doesFeatureMatch
} from '@badeball/cypress-cucumber-preprocessor';

beforeEach(() => {
  // This is only run for Cucumber-type specs
  if (isFeature() && doesFeatureMatch('@log')) {
    console.log('*** LOG ***');
  }
});
```

### Test code to reproduce

https://github.com/apehead/issue-cypress-cucumber-preprocessor

### Versions

* **Cypress version**: v12 (also verified the same behavior in v10 and v11)
* **Preprocessor version**: v17 (also verified the same behavior in v15 and v16)
* **Node version**: v18 (also verified the same behavior in v16)

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#bug-reports).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by apehead** [comment]
Hi @badeball! Thank you for maintaining this project :pray: 

If you need more information about the issue, please let me know.
I've tested different versions of the preprocessor, Cypress, and Node.js as described above. I'm not sure if I missed something or if the behavior before v15 was intentional.

Either way, I'm happy to help if needed.

I want to mention that I have tested the combination of preprocessor v14 with Cypress v12 and it works as expected, although the support for Cypress v12 was introduced in v15.
I hope this information is helpful for troubleshooting the issue.

**Comment by badeball** [comment]
Hi @apehead, thanks for reporting this issue. I've fixed this with v17.2.1.

**Comment by apehead** [comment]
Thanks for fixing this 🙂
